types/block_spec.md

```diff
@@ -0,0 +1,124 @@
+	    validator.ValidateBasic()
+		  validate not nil
+		  validator.PubKey not nil
+		  validator.VotingPower >= 0
+		  validator.Address == correct size
+	  // apply ValidateBasic to the proposer field:
+	  sh.Validators.Proposer.ValidateBasic()
+		validate not nil
+		validator.PubKey not nil
+		validator.VotingPower >= 0
+		validator.Address == correct size
+    Assert that SignedHeader.Validators.Hash() == SignedHeader.AggregatorsHash
+    Assert that len(SignedHeader.Commit.Signatures) == 1 (Exactly one signer check)
+	Verify the 1 signature
+  Data.ValidateBasic() // always passes
+  // make sure the SignedHeader's DataHash is equal to the hash of the actual data in the block.
+  Data.Hash() == SignedHeader.DataHash
+```
+
+## Verification Against Previous Block
+
+```go
+// (Block.Verify is currenty unimplemented)
+Block.Verify()
+  SignedHeader.Verify(untrustH *SignedHeader)
+    // basic validation removed in #1231, because go-header already validates it
+    //untrustH.ValidateBasic()
+	Header.Verify(untrustH *SignedHeader)
+	  if untrustH.Height == h.Height + 1, then apply the following check:
+	    untrstH.AggregatorsHash[:], h.NextAggregatorsHash[:]
+	if untrustH.Height > h.Height + 1:
+	  soft verification failure	
+	// We should know they're adjacent now,
+	// verify the link to previous.
+	untrustH.LastHeaderHash == h.Header.Hash()
+	// Verify LastCommit hash
+	untrustH.LastCommitHash == sh.Commit.GetCommitHash(...)
+	
+```
+
+## [Block](https://github.com/rollkit/rollkit/blob/main/types/block.go#L26)
+
+| **Field Name** | **Valid State**                         | **Validation**                     |
+|----------------|-----------------------------------------|------------------------------------|
+| SignedHeader   | Header of the block, signed by proposer | (See SignedHeader)                 |
+| Data           | Transaction data of the block           | Data.Hash == SignedHeader.DataHash |
+
+## [SignedHeader](https://github.com/rollkit/rollkit/blob/main/types/signed_header.go#L16)
+
+| **Field Name** | **Valid State**                                                          | **Validation**                                                                              |
+|----------------|--------------------------------------------------------------------------|---------------------------------------------------------------------------------------------|
+| Header         | Valid header for the block                                               | `Header` passes `ValidateBasic()` and `Verify()`                                            |
+| Commit         | 1 valid signature from the expected proposer                             | `Commit` passes `ValidateBasic()`, with additional checks in `SignedHeader.ValidateBasic()` |
+| Validators     | Array of Aggregators, should be length 1 (or zero for based rollup case) | `Validators` passes `ValidateBasic()`                                                       |
```

Comment:
should be at least length 1? Otherwise why is it an array.

Reply:
issue created
https://github.com/rollkit/rollkit/issues/1273